provisioner/puppet-masterless/provisioner_test.go

```diff
@@ -177,3 +179,63 @@ func TestProvisionerPrepare_facterFacts(t *testing.T) {
 		t.Fatalf("err: Default facts are not set in the Puppet provisioner!")
 	}
 }
+
+func TestProvisionerPrepare_options(t *testing.T) {
+	config := testConfig()
+
+	delete(config, "options")
+	p := new(Provisioner)
+	err := p.Prepare(config)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	// Test with malformed fact
+	config["options"] = "{{}}"
+	p = new(Provisioner)
+	err = p.Prepare(config)
+	if err == nil {
+		t.Fatal("should be an error")
+	}
+
+	config["options"] = []string{
+		"arg",
+	}
+
+	p = new(Provisioner)
+	err = p.Prepare(config)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+}
+
+func TestProvisionerProvision_options(t *testing.T) {
+	config := testConfig()
+	ui := &packer.MachineReadableUi{
+		Writer: ioutil.Discard,
+	}
+	comm := new(packer.MockCommunicator)
+
+	options := []string{
+		"--some-arg=yup",
+		"--some-other-arg",
+	}
+	config["options"] = options
```

Comment:
Hmm, not much I can really test for in the absence of values, other than maybe an extraneous space.